### ISSUE TITLE
Bengali / Khmer / Gujarati / Odia / Hindi regression?

### ISSUE THREAD

**Issue by masaccio** [jquast/wcwidth#126]
I recently updated from 0.2.6 to 0.2.13 and I have some tests breaking in a package that uses `wcswidth`. The following test fails every check in 0.2.13 but passes in 0.2.6:

``` python
from wcwidth import wcswidth

def check(length, value):
    if wcswidth(value) == length:
        print(value, "OK")
    else:
        print(value, "FAIL")

check(10, '"আবখাজিয়া"')
check(12, '"আফগানিস্তান"')
check(10, '"আলবেনিয়া"')
check(10, '"अबख़ाज़िया"')
check(11, '"ඇෆ්ගනිස්ථානය"')
check(10, '"ඇල්බේනියාව"')
check(12, '"अफ़ग़ानिस्तान"')
check(12, '"ஆக்கானித்தான்"')
check(14, '"អាហ្វហ្គានីស្ថាន"')
check(12, '"અફગાનિસ્તાન"')
check(9, '"અલ્બેનિયા"')
check(12, '"ଆଫାଗାନିସ୍ତାନ୍"')
check(13, '"गन्धार, अश्वक"')
check(10, '"अल्बानिया"')
```

Aligning some ASCII text in my terminal, I believe that the check lengths are correct:

<img width="222" alt="Screenshot 2024-05-15 at 15 13 55" src="https://github.com/jquast/wcwidth/assets/6770005/dda8567a-eca7-4671-bb55-e5487a248231">

**Comment by jquast** [comment]
@masaccio which terminal?

**Comment by jquast** [comment]
https://github.com/jquast/wcwidth/pull/91#issuecomment-1785693243 related issue and comment about the change

**Comment by masaccio** [comment]
This was iTerm2 on a Mac with `TERM=xterm-256color`

**Comment by masaccio** [comment]
This is the text I expect to see aligned - https://raw.githubusercontent.com/masaccio/compact-json/main/tests/data/test-issue-4.ref-1.json

Though in my browser it's not aligned so I don't know what the right answer is.

**Comment by jquast** [comment]
I will say that I also use iTerm2, and that it is not a great indicator of multilanguage support. I have since authored a testing and reporting tool, [ucs-detect](https://github.com/jquast/ucs-detect/), and have [published results](https://ucs-detect.readthedocs.io/results.html) for ~27 terminals.

The following terminals match this library's measurements for Hindi:
- WezTerm
- QTerminal
- mlterm
- Kovid Goyal's kitty

The other ~23 terminals, including iTerm2, do not. iTerm2 gets an overall score of "B" rating for LANG score while the ones listed above get A's.

Some of them are systematic errors and I may create bug reports for their respective projects. However, languages like Sanskrit and Hindi are very excessive with combining characters (Category codes Mc and Mn), and, strictly following the Unicode Specifications, as these 4 terminals and this library do, may result in so much "squeezing" to be totally **illegible**!

On your findings of the browser, I have found that they do not make the effort to align by column as a terminal is expected to (see screenshots in https://github.com/jquast/wcwidth/issues/123#issuecomment-2028115594)

I have authored a dummy "check" function to display a sequence where '|' should align,
```python
def check(n, phrase):
     print('|'+(' '*wcwidth.wcswidth(phrase))+'|'+'\n'+'|'+phrase+'|\n')
```

And these are the results for iTerm (left) and WezTerm (right)

![image](https://github.com/jquast/wcwidth/assets/972584/573e51a5-a718-4452-9df1-a154a70302a8)

I don't know Devanagari enough to say for sure, I would say that iTerm2 appears to fail to correctly combine characters of category Mc and Mn, while wezterm does combine them but also sometimes reduces the font size to accommodate their expected with and maybe some combining characters are also poorly aligned
